### PR TITLE
Expose the whole tls.Config setting

### PR DESCRIPTION
**Pull request by fbiville**

### CODE REVIEW

File: neo4j/internal/connector/connector.go [hunk 1]
```diff
@@ -69,19 +71,28 @@ func (c Connector) Connect(ctx context.Context, address string, boltLogger log.B
...
-	config := tls.Config{InsecureSkipVerify: c.SkipVerify, RootCAs: c.RootCAs, ServerName: serverName}
-	tlsconn := tls.Client(conn, &config)
-	err = tlsconn.HandshakeContext(ctx)
+	tlsConn := tls.Client(conn, c.tlsConfig(serverName))
+	err = tlsConn.HandshakeContext(ctx)
 	if err != nil {
 		if err == io.EOF {
 			// Give a bit nicer error message
-			err = errors.New("Remote end closed the connection, check that TLS is enabled on the server")
+			err = errors.New("remote end closed the connection, check that TLS is enabled on the server")
```
Comment (bigmontz) [inline neo4j/internal/connector/connector.go:79]: Was it intentional ?

File: neo4j/config.go [hunk 2]
```diff
@@ -36,7 +37,19 @@ type Config struct {
...
+	//
+	// The configuration is only used for URI schemes 'bolt+s', 'bolt+ssc',
+	// 'neo4j+s' and 'neo4j+ssc'.
+	//
+	// The SkipVerify attribute of TlsConfig is always derived from the initial URI scheme.
+	// The ServerName attribute of TlsConfig is always derived from the initial URI host.
+
+	// The RootCAs attribute of this TlsConfig has higher precedence than the
+	// attribute set on the enclosing Config.
+	TlsConfig *tls.Config
```
Comment (bigmontz) [inline neo4j/config.go:52]: Is it just ignore when use `bolt` and `neo4j`?

**Review by bigmontz — APPROVED** [review]
⛓️

File: neo4j/internal/connector/connector.go (hunk 1 above)
Comment (fbiville) [reply neo4j/internal/connector/connector.go:79]: yes, Go errors should not start with uppercase, so I finally gave in to the IDE complaint and fixed it

File: neo4j/config.go (hunk 2 above)
Comment (fbiville) [reply neo4j/config.go:52]: yes, see line 64 of `neo4j/internal/connector/connector.go` (line 62 in the old version)

**Comment by fbiville** [comment]
Identified flaky test failing. Will investigate after this PR. Merging now.

**Pull request merged**